### PR TITLE
Splitting i client request

### PR DESCRIPTION
**Pull request by toppggg**
I have split IClientRequest into separate interfaces for each of the users, so that we can implement classes that only uses a part of the full interface.

### CODE REVIEW

File: src/Model/Database/DatabaseServices.cs [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
 using Result = Model.Result.Result;
-public class DatabaseServices {
+using Model.Result;
+using Model.Answer;
+using System.Collections.Generic;
+
+public class DatabaseServices : IDatabase {
 
     private string databasePath = "./surveyDatabase/";
+    private string resultsPath;
```
Comment (jonasreholt) [inline src/Model/Database/DatabaseServices.cs:16]: Is it intentional that all surveys will save their results to the same file?

File: src/Model/FrontEndAPI/FrontEndMainMenu.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            }
+            return true;
+        }
+        catch (Exception) {
+            return false;
+        } 
+    }
+
+    public IReadOnlySurvey GetSurvey(int surveyId) {
+        return databaseService.GetSurvey(surveyId);
```
Comment (jonasreholt) [inline src/Model/FrontEndAPI/FrontEndMainMenu.cs:31]:
```suggestion
    public IReadOnlySurvey? GetSurvey(int pincode) {
        return databaseService.GetSurvey(pincode);
```

File: src/Model/tmp_Moc/FrontEndMainMenu.cs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 namespace Model.FrontEndAPI;
 using Model.Survey;
 
-public static class FrontEndMainMenu
+public static class JonasFrontEndMainMenu
```
Comment (jonasreholt) [inline src/Model/tmp_Moc/FrontEndMainMenu.cs:4]:
```suggestion
public static class FrontEndMainMenuMock
```

**Review by jonasreholt — APPROVED** [review]
LGTM

File: src/Model/Database/DatabaseServices.cs (hunk 1 above)
Comment (toppggg) [reply src/Model/Database/DatabaseServices.cs:16]: I think that we'll just have a single table with results, where you then query results from a specific survey. When extracted, you just 'select * Results where surveyId = requestedId'

**Review by jonasreholt — APPROVED** [review]

**Pull request merged**